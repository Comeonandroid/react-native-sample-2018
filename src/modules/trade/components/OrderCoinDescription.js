import React, { PureComponent } from 'react'
import styled, { css } from 'styled-components'
import { ifProp, prop } from 'styled-tools'
import PropTypes from 'prop-types'

import { t } from '../../../locales/i18n'
import CustomIcon from '../../../utils/icon'

import { View, Roboto } from '../../../elements'

class OrderCoinDescription extends PureComponent {
  render() {
    const { price, percentageChange, trend, vol, high, low, contract } = this.props
    return (
      <Container trend={trend} no-padding={'true'}>
        <Cell max-width='50%'>
          <Price>{price + ' '}
            <Percentage trend={trend}>
              ({percentageChange}%
              <Arrow name='arrow-up' size={12} trend={trend} />)
            </Percentage>
          </Price>
          <FiatPrice numberOfLines={1}>{t('trade.contract')}: {contract}</FiatPrice>
        </Cell>
        <Cell flex-direction={'row'} no-padding={!!true}>
        </Cell>
        <Cell margin-left='auto'>
          <View>
            <MarketLabel>{t('trade.vol')} {vol}</MarketLabel>
          </View>
          <View>
            <MarketLabel>{t('trade.high')} {high}</MarketLabel>
          </View>
          <View>
            <MarketLabel>{t('trade.low')} {low}</MarketLabel>
          </View>
        </Cell>
      </Container>
    )
  }
}

const Container = styled(View)`
  display: flex;
  borderColor: ${ifProp({ trend: true }, prop('theme.colors.success'), prop('theme.colors.danger'))}
  borderTopWidth: 1px;
  background-color: ${prop('theme.colors.simpleButtonBackground')};
  flex-direction: row;
  align-items: center;
  padding: 5px 0 5px 10px;
`

const Cell = styled(View)`
  margin-left: ${ifProp({ 'margin-left': 'auto' }, prop('margin-left'), 0)};
  align-items: ${ifProp({ center: 'true' }, 'center', 'flex-start')};
  padding: ${ifProp('no-padding', 0, '0 10px ')};
  display: flex;
  flex-direction: ${ifProp({ 'flex-direction': 'row' }, 'row', 'column')}
  width: ${ifProp('max-width', prop('max-width'),  'auto')};
`

const TrendText = styled(Roboto)`
  color: ${ifProp({ trend: true }, prop('theme.colors.success'), prop('theme.colors.danger'))}
`

const Price = styled(TrendText)`
  font-size: 15px;
  line-height: 16px;
  color: ${prop('theme.colors.accountHeaderColor')}
`

const FiatPrice = styled(Roboto)`
  font-size: 12px;
  color: ${prop('theme.colors.accountHeaderColor')}
`
const Arrow = styled(CustomIcon)`
  margin-right: 5px;
  ${ifProp('trend',
    css`
      color: ${prop('theme.colors.success')};
    `,
    css`
      color: ${prop('theme.colors.danger')};
      transform: rotate(180deg);
    `
  )}
`

const Percentage = styled(TrendText)`
  font-size: 14px;
  font-weight: bold;
`

const MarketLabel = styled(Roboto)`
  color: ${prop('theme.colors.tradeBrightTextColor')}
  font-size: 13px;
  line-height: 14px;
`


OrderCoinDescription.propTypes = {
  price: PropTypes.number,
  trend: PropTypes.bool,
  percentageChange: PropTypes.number,
  vol: PropTypes.number,
  high: PropTypes.number,
  low: PropTypes.number,
  contract: PropTypes.string
}

export default OrderCoinDescription
