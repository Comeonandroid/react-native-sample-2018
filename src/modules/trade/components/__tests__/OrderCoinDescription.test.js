import 'react-native'
import React from 'react'

import { itBehavesLike } from '../../../../specs/shared_examples/index'

import OrderCoinDescription from '../OrderCoinDescription'

const setup = propOverrides => {
  const props = Object.assign({
    price: 0.1112323,
    trend: false,
    percentageChange: 0.18,
    vol: 64.23232,
    high: 0.01212,
    low: 0.011111,
    fiatPrice: 10.22,
    currency: '$',
    logoUrl: 'https://s3-ap-southeast-1.amazonaws.com/tokenjar/zrx.png'
  }, propOverrides)

  const component = <OrderCoinDescription {...props} />

  return {
    props,
    component
  }
}

describe('<OrderCoinDescription />', () => {
  const { component } = setup()
  itBehavesLike('aThemedComponent', { subject: component })

  describe('trend: true', () => {
    const { component } = setup({ trend: true })
    itBehavesLike('aThemedComponent', { subject: component })
  })
})
