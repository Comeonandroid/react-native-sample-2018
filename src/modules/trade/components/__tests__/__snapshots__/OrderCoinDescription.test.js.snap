// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderCoinDescription /> dark: renders correctly 1`] = `
<View
  no-padding="true"
  style={
    Array [
      Object {},
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#000",
          "borderColor": "#f25c74",
          "borderTopWidth": 1,
          "display": "flex",
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingLeft": 10,
          "paddingRight": 0,
          "paddingTop": 5,
        },
        undefined,
      ],
    ]
  }
  trend={false}
>
  <View
    center="true"
    style={
      Array [
        Object {},
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "column",
            "marginLeft": 0,
            "paddingBottom": 0,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 0,
          },
          undefined,
        ],
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontFamily": "Roboto Condensed",
            "fontWeight": "400",
          },
          Array [
            Object {
              "color": "#f25c74",
            },
            Array [
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
                "lineHeight": 16,
              },
              undefined,
            ],
          ],
        ]
      }
      trend={false}
    >
      0.1112323
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontFamily": "Roboto Condensed",
            "fontWeight": "400",
          },
          Array [
            Object {
              "color": "#fff",
              "fontSize": 12,
            },
            undefined,
          ],
        ]
      }
    >
      $
       
      10.22
    </Text>
  </View>
  <View
    flex-direction="row"
    no-padding={true}
    style={
      Array [
        Object {},
        Array [
          Object {
            "alignItems": "flex-start",
            "display": "flex",
            "flexDirection": "row",
            "marginLeft": 0,
            "paddingBottom": 0,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 0,
          },
          undefined,
        ],
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Array [
            Object {
              "color": "#fcfcfc",
            },
            Array [
              Object {
                "color": "#f25c74",
                "marginRight": 5,
                "transform": Array [
                  Object {
                    "rotate": "180deg",
                  },
                ],
              },
              undefined,
            ],
          ],
          Object {
            "fontFamily": "icomoon",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
      trend={false}
    >
      î¤”
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontFamily": "Roboto Condensed",
            "fontWeight": "400",
          },
          Array [
            Object {
              "color": "#f25c74",
            },
            Array [
              Object {
                "fontSize": 14,
                "fontWeight": "bold",
              },
              undefined,
            ],
          ],
        ]
      }
      trend={false}
    >
      0.18
      %
    </Text>
  </View>
  <View
    margin-left="auto"
    no-padding="true"
    style={
      Array [
        Object {},
        Array [
          Object {
            "alignItems": "flex-start",
            "display": "flex",
            "flexDirection": "column",
            "marginLeft": "auto",
            "paddingBottom": 0,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 0,
          },
          undefined,
        ],
      ]
    }
  >
    <Image
      source={
        Object {
          "uri": "https://s3-ap-southeast-1.amazonaws.com/tokenjar/zrx.png",
        }
      }
      style={
        Array [
          Object {
            "height": 30,
            "width": 30,
          },
          undefined,
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {},
        Array [
          Object {
            "alignItems": "flex-start",
            "display": "flex",
            "flexDirection": "column",
            "marginLeft": 0,
            "paddingBottom": 0,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 0,
          },
          undefined,
        ],
      ]
    }
  >
    <View
      style={
        Array [
          Object {},
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": "Roboto Condensed",
              "fontWeight": "400",
            },
            Array [
              Object {
                "color": "#c8c8c8",
                "fontSize": 13,
                "lineHeight": 14,
              },
              undefined,
            ],
          ]
        }
      >
        Vol.
         
        64.23232
      </Text>
    </View>
    <View
      style={
        Array [
          Object {},
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": "Roboto Condensed",
              "fontWeight": "400",
            },
            Array [
              Object {
                "color": "#c8c8c8",
                "fontSize": 13,
                "lineHeight": 14,
              },
              undefined,
            ],
          ]
        }
      >
        High
         
        0.01212
      </Text>
    </View>
    <View
      style={
        Array [
          Object {},
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": "Roboto Condensed",
              "fontWeight": "400",
            },
            Array [
              Object {
                "color": "#c8c8c8",
                "fontSize": 13,
                "lineHeight": 14,
              },
              undefined,
            ],
          ]
        }
      >
        Low
         
        0.011111
      </Text>
    </View>
  </View>
</View>
`;

exports[`<OrderCoinDescription /> light: renders correctly 1`] = `
<View
  no-padding="true"
  style={
    Array [
      Object {},
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#5d5959",
          "borderColor": "#f25c74",
          "borderTopWidth": 1,
          "display": "flex",
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingLeft": 10,
          "paddingRight": 0,
          "paddingTop": 5,
        },
        undefined,
      ],
    ]
  }
  trend={false}
>
  <View
    center="true"
    style={
      Array [
        Object {},
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "column",
            "marginLeft": 0,
            "paddingBottom": 0,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 0,
          },
          undefined,
        ],
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontFamily": "Roboto Condensed",
            "fontWeight": "400",
          },
          Array [
            Object {
              "color": "#f25c74",
            },
            Array [
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
                "lineHeight": 16,
              },
              undefined,
            ],
          ],
        ]
      }
      trend={false}
    >
      0.1112323
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontFamily": "Roboto Condensed",
            "fontWeight": "400",
          },
          Array [
            Object {
              "color": "#fff",
              "fontSize": 12,
            },
            undefined,
          ],
        ]
      }
    >
      $
       
      10.22
    </Text>
  </View>
  <View
    flex-direction="row"
    no-padding={true}
    style={
      Array [
        Object {},
        Array [
          Object {
            "alignItems": "flex-start",
            "display": "flex",
            "flexDirection": "row",
            "marginLeft": 0,
            "paddingBottom": 0,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 0,
          },
          undefined,
        ],
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Array [
            Object {
              "color": "#4b4c4d",
            },
            Array [
              Object {
                "color": "#f25c74",
                "marginRight": 5,
                "transform": Array [
                  Object {
                    "rotate": "180deg",
                  },
                ],
              },
              undefined,
            ],
          ],
          Object {
            "fontFamily": "icomoon",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
      trend={false}
    >
      î¤”
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontFamily": "Roboto Condensed",
            "fontWeight": "400",
          },
          Array [
            Object {
              "color": "#f25c74",
            },
            Array [
              Object {
                "fontSize": 14,
                "fontWeight": "bold",
              },
              undefined,
            ],
          ],
        ]
      }
      trend={false}
    >
      0.18
      %
    </Text>
  </View>
  <View
    margin-left="auto"
    no-padding="true"
    style={
      Array [
        Object {},
        Array [
          Object {
            "alignItems": "flex-start",
            "display": "flex",
            "flexDirection": "column",
            "marginLeft": "auto",
            "paddingBottom": 0,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 0,
          },
          undefined,
        ],
      ]
    }
  >
    <Image
      source={
        Object {
          "uri": "https://s3-ap-southeast-1.amazonaws.com/tokenjar/zrx.png",
        }
      }
      style={
        Array [
          Object {
            "height": 30,
            "width": 30,
          },
          undefined,
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {},
        Array [
          Object {
            "alignItems": "flex-start",
            "display": "flex",
            "flexDirection": "column",
            "marginLeft": 0,
            "paddingBottom": 0,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 0,
          },
          undefined,
        ],
      ]
    }
  >
    <View
      style={
        Array [
          Object {},
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": "Roboto Condensed",
              "fontWeight": "400",
            },
            Array [
              Object {
                "color": "#fff",
                "fontSize": 13,
                "lineHeight": 14,
              },
              undefined,
            ],
          ]
        }
      >
        Vol.
         
        64.23232
      </Text>
    </View>
    <View
      style={
        Array [
          Object {},
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": "Roboto Condensed",
              "fontWeight": "400",
            },
            Array [
              Object {
                "color": "#fff",
                "fontSize": 13,
                "lineHeight": 14,
              },
              undefined,
            ],
          ]
        }
      >
        High
         
        0.01212
      </Text>
    </View>
    <View
      style={
        Array [
          Object {},
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": "Roboto Condensed",
              "fontWeight": "400",
            },
            Array [
              Object {
                "color": "#fff",
                "fontSize": 13,
                "lineHeight": 14,
              },
              undefined,
            ],
          ]
        }
      >
        Low
         
        0.011111
      </Text>
    </View>
  </View>
</View>
`;

exports[`<OrderCoinDescription /> trend: true dark: renders correctly 1`] = `
<View
  no-padding="true"
  style={
    Array [
      Object {},
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#000",
          "borderColor": "#2a986f",
          "borderTopWidth": 1,
          "display": "flex",
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingLeft": 10,
          "paddingRight": 0,
          "paddingTop": 5,
        },
        undefined,
      ],
    ]
  }
  trend={true}
>
  <View
    center="true"
    style={
      Array [
        Object {},
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "column",
            "marginLeft": 0,
            "paddingBottom": 0,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 0,
          },
          undefined,
        ],
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontFamily": "Roboto Condensed",
            "fontWeight": "400",
          },
          Array [
            Object {
              "color": "#2a986f",
            },
            Array [
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
                "lineHeight": 16,
              },
              undefined,
            ],
          ],
        ]
      }
      trend={true}
    >
      0.1112323
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontFamily": "Roboto Condensed",
            "fontWeight": "400",
          },
          Array [
            Object {
              "color": "#fff",
              "fontSize": 12,
            },
            undefined,
          ],
        ]
      }
    >
      $
       
      10.22
    </Text>
  </View>
  <View
    flex-direction="row"
    no-padding={true}
    style={
      Array [
        Object {},
        Array [
          Object {
            "alignItems": "flex-start",
            "display": "flex",
            "flexDirection": "row",
            "marginLeft": 0,
            "paddingBottom": 0,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 0,
          },
          undefined,
        ],
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Array [
            Object {
              "color": "#fcfcfc",
            },
            Array [
              Object {
                "color": "#2a986f",
                "marginRight": 5,
              },
              undefined,
            ],
          ],
          Object {
            "fontFamily": "icomoon",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
      trend={true}
    >
      î¤”
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontFamily": "Roboto Condensed",
            "fontWeight": "400",
          },
          Array [
            Object {
              "color": "#2a986f",
            },
            Array [
              Object {
                "fontSize": 14,
                "fontWeight": "bold",
              },
              undefined,
            ],
          ],
        ]
      }
      trend={true}
    >
      0.18
      %
    </Text>
  </View>
  <View
    margin-left="auto"
    no-padding="true"
    style={
      Array [
        Object {},
        Array [
          Object {
            "alignItems": "flex-start",
            "display": "flex",
            "flexDirection": "column",
            "marginLeft": "auto",
            "paddingBottom": 0,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 0,
          },
          undefined,
        ],
      ]
    }
  >
    <Image
      source={
        Object {
          "uri": "https://s3-ap-southeast-1.amazonaws.com/tokenjar/zrx.png",
        }
      }
      style={
        Array [
          Object {
            "height": 30,
            "width": 30,
          },
          undefined,
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {},
        Array [
          Object {
            "alignItems": "flex-start",
            "display": "flex",
            "flexDirection": "column",
            "marginLeft": 0,
            "paddingBottom": 0,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 0,
          },
          undefined,
        ],
      ]
    }
  >
    <View
      style={
        Array [
          Object {},
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": "Roboto Condensed",
              "fontWeight": "400",
            },
            Array [
              Object {
                "color": "#c8c8c8",
                "fontSize": 13,
                "lineHeight": 14,
              },
              undefined,
            ],
          ]
        }
      >
        Vol.
         
        64.23232
      </Text>
    </View>
    <View
      style={
        Array [
          Object {},
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": "Roboto Condensed",
              "fontWeight": "400",
            },
            Array [
              Object {
                "color": "#c8c8c8",
                "fontSize": 13,
                "lineHeight": 14,
              },
              undefined,
            ],
          ]
        }
      >
        High
         
        0.01212
      </Text>
    </View>
    <View
      style={
        Array [
          Object {},
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": "Roboto Condensed",
              "fontWeight": "400",
            },
            Array [
              Object {
                "color": "#c8c8c8",
                "fontSize": 13,
                "lineHeight": 14,
              },
              undefined,
            ],
          ]
        }
      >
        Low
         
        0.011111
      </Text>
    </View>
  </View>
</View>
`;

exports[`<OrderCoinDescription /> trend: true light: renders correctly 1`] = `
<View
  no-padding="true"
  style={
    Array [
      Object {},
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#5d5959",
          "borderColor": "#2a986f",
          "borderTopWidth": 1,
          "display": "flex",
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingLeft": 10,
          "paddingRight": 0,
          "paddingTop": 5,
        },
        undefined,
      ],
    ]
  }
  trend={true}
>
  <View
    center="true"
    style={
      Array [
        Object {},
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "column",
            "marginLeft": 0,
            "paddingBottom": 0,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 0,
          },
          undefined,
        ],
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontFamily": "Roboto Condensed",
            "fontWeight": "400",
          },
          Array [
            Object {
              "color": "#2a986f",
            },
            Array [
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
                "lineHeight": 16,
              },
              undefined,
            ],
          ],
        ]
      }
      trend={true}
    >
      0.1112323
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontFamily": "Roboto Condensed",
            "fontWeight": "400",
          },
          Array [
            Object {
              "color": "#fff",
              "fontSize": 12,
            },
            undefined,
          ],
        ]
      }
    >
      $
       
      10.22
    </Text>
  </View>
  <View
    flex-direction="row"
    no-padding={true}
    style={
      Array [
        Object {},
        Array [
          Object {
            "alignItems": "flex-start",
            "display": "flex",
            "flexDirection": "row",
            "marginLeft": 0,
            "paddingBottom": 0,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 0,
          },
          undefined,
        ],
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Array [
            Object {
              "color": "#4b4c4d",
            },
            Array [
              Object {
                "color": "#2a986f",
                "marginRight": 5,
              },
              undefined,
            ],
          ],
          Object {
            "fontFamily": "icomoon",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
      trend={true}
    >
      î¤”
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontFamily": "Roboto Condensed",
            "fontWeight": "400",
          },
          Array [
            Object {
              "color": "#2a986f",
            },
            Array [
              Object {
                "fontSize": 14,
                "fontWeight": "bold",
              },
              undefined,
            ],
          ],
        ]
      }
      trend={true}
    >
      0.18
      %
    </Text>
  </View>
  <View
    margin-left="auto"
    no-padding="true"
    style={
      Array [
        Object {},
        Array [
          Object {
            "alignItems": "flex-start",
            "display": "flex",
            "flexDirection": "column",
            "marginLeft": "auto",
            "paddingBottom": 0,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 0,
          },
          undefined,
        ],
      ]
    }
  >
    <Image
      source={
        Object {
          "uri": "https://s3-ap-southeast-1.amazonaws.com/tokenjar/zrx.png",
        }
      }
      style={
        Array [
          Object {
            "height": 30,
            "width": 30,
          },
          undefined,
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {},
        Array [
          Object {
            "alignItems": "flex-start",
            "display": "flex",
            "flexDirection": "column",
            "marginLeft": 0,
            "paddingBottom": 0,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 0,
          },
          undefined,
        ],
      ]
    }
  >
    <View
      style={
        Array [
          Object {},
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": "Roboto Condensed",
              "fontWeight": "400",
            },
            Array [
              Object {
                "color": "#fff",
                "fontSize": 13,
                "lineHeight": 14,
              },
              undefined,
            ],
          ]
        }
      >
        Vol.
         
        64.23232
      </Text>
    </View>
    <View
      style={
        Array [
          Object {},
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": "Roboto Condensed",
              "fontWeight": "400",
            },
            Array [
              Object {
                "color": "#fff",
                "fontSize": 13,
                "lineHeight": 14,
              },
              undefined,
            ],
          ]
        }
      >
        High
         
        0.01212
      </Text>
    </View>
    <View
      style={
        Array [
          Object {},
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": "Roboto Condensed",
              "fontWeight": "400",
            },
            Array [
              Object {
                "color": "#fff",
                "fontSize": 13,
                "lineHeight": 14,
              },
              undefined,
            ],
          ]
        }
      >
        Low
         
        0.011111
      </Text>
    </View>
  </View>
</View>
`;
